frontend/src/pages/ActivityReport/Pages/components/GoalPicker.js

```diff
@@ -105,10 +111,54 @@ const GoalPicker = ({
       const topics = await getTopics();
       setTopicOptions(topics);
     }
-
     fetchTopics();
   }, []);
 
+  // Fetch citations for the goal if the source is CLASS or RANs.
+  useEffect(() => {
```

Comment:
```suggestion
  useDeepCompareEffect(() => {
```

Reply:
you'll need to import that lib as well @AdamAdHocTeam

Reply:
done.